### DISCUSSION TITLE
Modular Switch with different Modules and Port countme

### DISCUSSION THREAD

**Discussion by marcohald** [netbox-community/netbox#8325] [Q&A] [answered]
What is the best way to create a Modular Switch with different modules in it that have different Port counts.
The Switches should be both the same Device Type as they are the same chassis.
One possibility would be to add the modules as a Device bay, but the documentation states that this is the wrong way https://github.com/netbox-community/netbox/pull/2793 .
Do I need to create a different Device Type for every module configuration that we use ?

**Comment by jeremystretch** [comment, marked as answer]
In the current release, the recommended approach is to simply add the relevant interfaces directly to the device, and do add any modules as inventory items if needed.

NetBox v3.2 will introduce modules & module bays to more accurately model these components, and ease the addition of their associated interfaces (see #7844).

**Reply by marcohald** [reply]
Perfect that is what I'm looking for.
Is the Naming of the Module Interfaces with Characters already implemented (mentioned here https://github.com/netbox-community/netbox/issues/7844#issuecomment-997931166 ) ?
Because our HP Switches Label the Module Slots with Characters beginning from A.
So the Port Name would be A24 or C3

**Reply by jeremystretch** [reply]
Yes; the component names can be templated to replace part of their name with the index assigned to the module bay into which their parent module is installed.
